x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -259,17 +269,24 @@ private static void checkBinaryComparison(LogicalPlan p, Set<Failure> failures)
         });
     }
 
-    private void gatherMetrics(LogicalPlan plan) {
-        BitSet b = new BitSet(FeatureMetric.values().length);
+    private void gatherMetrics(LogicalPlan plan, BitSet b) {
         plan.forEachDown(p -> {
             if (p instanceof Dissect) {
                 b.set(DISSECT.ordinal());
+            } else if (p instanceof Enrich) {
+                b.set(ENRICH.ordinal());
             } else if (p instanceof Eval) {
                 b.set(EVAL.ordinal());
+            } else if (p instanceof EsRelation) {
+                b.set(FROM.ordinal());
             } else if (p instanceof Grok) {
                 b.set(GROK.ordinal());
-            } else if (p instanceof Limit) {
-                b.set(LIMIT.ordinal());
+            } else if (p instanceof MvExpand) {
+                b.set(MV_EXPAND.ordinal());
+            } else if (p instanceof Row) {
+                b.set(ROW.ordinal());
+            } else if (p instanceof ShowInfo || p instanceof ShowFunctions) {
+                b.set(SHOW.ordinal());
             } else if (p instanceof OrderBy) {
                 b.set(SORT.ordinal());
             } else if (p instanceof Aggregate) {
```

Comment:
Since this type of code is in two places, you could encapsulate it into one place.
To avoid the large if, add an utility method on FeatureMetric that iterates over a static Map, with the keys either a class or a `java.util.Predicate` that if it matches, returns the enum or even better sets the bitset.

```
set(LogicalPlan plan, BitSet bitset) {
  for (Entry<Predicate, Feature> entry : features.entrySet()) {
        if (entry.key.test(plan)) {
              bitset.set(entry.value().ordinal)
        }
  }
}
```